### DISCUSSION TITLE
Receiving 3 MQTT messages for each button press.

### DISCUSSION THREAD

**Discussion by michelcmorel** [HASwitchPlate/HASPone#97] [Q&A] [answered]
Every time I press a button on the switchplate (like toggle blueprint), I receive 3 MQTT messages, so I'm just wondering if this is the correct or default behaviour?  Is anyone else getting the same behaviour? 
Also is there an option to receive only 1 MQTT message for 1 button press, by any chance?
![Screenshot 2021-05-06 at 12 36 42](https://user-images.githubusercontent.com/4426933/117299549-96880980-ae70-11eb-967d-c0b98d7d7185.png)

**Comment by rebel7580** [comment, marked as answer]
Yes. The ON comes when you press the button, the OFF occurs when you release it  (remove your finger!) So if you press and hold a button you will see only the ON until you eventually release the button at which point you will see the OFF.

I'm not sure why we're getting the 0 along with the OFF...

**Reply by rebel7580** [reply]
In fact, there are 6 MQTT messages sent out. You're only subscribed to the json ones:
```
[+162203.234s] MQTT OUT: 'hasp/hasp01/state/p[1].b[1]' : 'ON'
[+162203.250s] MQTT OUT: 'hasp/hasp01/state/json' : '{"event_type":"button_short_press","event":"p[1].b[1]","value":"ON"}'

[+162203.406s] MQTT OUT: 'hasp/hasp01/state/p[1].b[1]' : 'OFF'
[+162203.406s] MQTT OUT: 'hasp/hasp01/state/json' : '{"event_type":"button_short_release","event":"p[1].b[1]","value":"OFF"}'
'
[+162203.453s] MQTT OUT: 'hasp/hasp01/state/p[1].b[1].val' : '0'
[+162203.453s] MQTT OUT: 'hasp/hasp01/state/json' : '{"event":"p[1].b[1].val","value":0}'
```
MQTT is a very light-weight protocol and "extra" messages aren't usually a big deal for a network traffic perspetive. Just detect the one(s) you need and ignore the rest.

**Reply by aderusha** [reply]
100% on all of the above.  For any button press on the screen, you'll see the HASPone device send out two button "ON" messages, one in a JSON object and one in the HASPone state namespace.  These are both sent because (as noted above) it costs nearly nothing to offer both options, and either option might be desirable depending on the use case.  You'll find that the provided blueprints use one or the other as needs require.

Then, when released you'll see two more messages with the "OFF" state being reported, just like above and for the same reasons.

Finally, you'll get a 5th and 6th message with the "value" of the button you just released.  This value is queried by the HASPone firmware on any button release event.  For normal buttons this will always be 0 and can be ignored.  For sliders, this will be the current value of the slider and can thus be used to control things.

**Reply by aderusha** [reply]
Also, there is no option to disable these messages as doing so will break the provided blueprints.  Each of those messages is used in some context, and because they cost nearly nothing to send in terms of traffic and overall load, there's little incentive to change that.

**Comment by michelcmorel** [comment]
Thank you both for your feedback. Yes that makes complete sense. I needed a way to receive only 1 mqtt message from nodered for when I press on the toggle button of the HaspOne switch plate screen (3 min timer button). I ended up using "hasp/master/state/p[6].b[4].val" so I get notified only once when the button is pressed so I can then toggle the icon on the HaspOne screen from NodeRed and trigger an Alexa timer for my cup of tea :) and there is a toggle subflow to invert the value so the next press it sends a routine to cancel the alarm and toggle the icon off on the screen. Works quite well :)
![Screenshot 2021-05-09 at 20 47 13](https://user-images.githubusercontent.com/4426933/117585022-285d7400-b108-11eb-8fe3-08d1fe359014.png)
